OWML.ModHelper.Menus/ModInputCombinationPopup.cs

```diff
@@ -0,0 +1,201 @@
+
+        protected override void Update()
+        {
+            if (_resetCommand != null && OWInput.IsNewlyPressed(_resetCommand, InputMode.All))
+            {
+                InvokeReset();
+                OnPopupReset.Invoke();
+            }
+            List<KeyCode> currentlyPressedKeys = new List<KeyCode>();
+            for (var code = ModInputLibrary.MinUsefulKey; code < ModInputLibrary.MaxUsefulKey; code++)
+            {
+                if (!(Enum.IsDefined(typeof(KeyCode), (KeyCode)code) && UnityEngine.Input.GetKey((KeyCode)code)))
+                {
+                    continue;
+                }
+                currentlyPressedKeys.Add((KeyCode)code);
+            }
+            if (currentlyPressedKeys.Count == 1)
+            {
+                if (_cancelCommand != null && _cancelCommand.IsNewlyPressed())
+                {
+                    InvokeCancel();
+                    return;
+                }
+                if (_okCommand != null && _okCommand.IsNewlyPressed())
+                {
+                    InvokeOk();
+                    return;
+                }
+                if (_resetCommand != null && _resetCommand.IsNewlyPressed())
+                {
+                    InvokeReset();
+                    return;
+                }
+                if (currentlyPressedKeys[0] == KeyCode.Mouse0 || currentlyPressedKeys[0] == KeyCode.Mouse1)
+                {
+                    return;
+                }
+            }
+            if (currentlyPressedKeys.Count < 8 && (currentlyPressedKeys.Count > _combination.Count 
+                || (currentlyPressedKeys.Count > 1 && _wasReleased)))
+            {
+                _wasReleased = false;
+                _combination = currentlyPressedKeys;
+                UpdateContents();
+            }
+            if (currentlyPressedKeys.Count == 0)
+            {
+                _wasReleased = true;
+            }
+        }
+
+        public override void Activate()
+        {
+            base.Activate();
+            Locator.GetMenuInputModule().SelectOnNextUpdate(null); //unselect buttons
+        }
+
+        private void AddKeySign(KeyCode key)
+        {
+            Layout.AddPictureAt(_inputHandler.Textures.KeyTexture(key), Layout.ChildCount, ModInputLibrary.ScaleDown);
+        }
+
+        private void UpdateContents()
+        {
+            Layout.Clear();
+            for (var j = 0; j < _combination.Count; j++)
+            {
+                AddKeySign(_combination[j]);
+                if (j < _combination.Count - 1)
+                {
+                    Layout.AddText("+");
+                }
+            }
+            Layout.UpdateState();
+        }
+
+        public void EnableMenu(bool value, string currentCombination)
+        {
+            if (value)
+            {
+                _combination.Clear();
+                foreach (var key in currentCombination.Split('+'))
+                {
+                    if (key != "")
```

Comment:
foreach (var key in currentCombination.Split('+').Where(key => key != "")

Reply:
Could also turn it into one-liner

Reply:
Although that sounds like going too far TBH